### PR TITLE
fix bug in maxpool

### PR DESCRIPTION
**Pull request by zhijxu-MS**
ksize and strides are not always NHWC, actual their formats are aligned
with input data's format

### CODE REVIEW

**Comment by zhijxu-MS** [comment]
MaxPoolWithArgmax may share same bug, will take a look in next PR

**Comment by zhijxu-MS** [comment]
> MaxPoolWithArgmax may share same bug, will take a look in next PR

**Comment by zhijxu-MS** [comment]
also SpaceToDepth may share same bug.

File: tf2onnx/onnx_opset/nn.py [hunk 1]
```diff
@@ -301,25 +301,28 @@ def _convert(cls, ctx, node, **kwargs):
...
             ctx.remove_input(node, node.input[1])
 
+        if node.is_nhwc():
+            kernel_shape_hw = kernel_shape_tf[1:3]
+            strides_hw = strides_tf[1:3]
+        elif node.is_nchw():
+            kernel_shape_hw = kernel_shape_tf[2:4]
+            strides_hw = strides_tf[2:4]
+        else:
+            logger.warning("unexpected data format, please check it")
```
Comment (nbcsm) [inline tf2onnx/onnx_opset/nn.py]: throw
and better error message: tf op name, attribute name, expected values, actual value.

File: tests/common.py [hunk 2]
```diff
@@ -22,6 +23,7 @@
     "check_tf_min_version",
     "check_tf_max_version",
     "skip_tf_versions",
+    "skip_if_tf_cpu",
```
Comment (nbcsm) [inline tests/common.py]: skip_tf_cpu? to align with existing names

File: tf2onnx/graph.py [hunk 3]
```diff
@@ -136,9 +136,13 @@ def data_format(self, val):
         self.set_attr("data_format", val)
 
     def is_nhwc(self):
-        """Return True if node is in NCHW format."""
+        """Return True if node is in NHWC format."""
         return self.data_format == "NHWC"
 
+    def is_nchw(self):
```
Comment (nbcsm) [inline tf2onnx/graph.py]: not used? remove?

Comment (zhijxu-MS) [reply tf2onnx/graph.py]: done.

**Pull request merged**